src/components/base/BaseInput.vue

```diff
@@ -24,6 +31,48 @@ export default defineComponent({
       default: false,
     },
   },
+  data() {
+    return {
+      isPasswordVisible: false,
+      isdateInputText: false,
+    };
+  },
+  updated() {
+    if (this.type === 'date' && this.modelValue.length > 0) {
+      this.isdateInputText = true;
+    }
+  },
+  computed: {
+    computedType() {
+      if (
+        (this.type === 'date' && !this.isdateInputText) ||
+        (this.type === 'password' && this.isPasswordVisible)
+      ) {
+        return 'text';
+      } else return this.type;
+    },
+    inputClass() {
+      return {
+        'input__field--password': this.type === 'password',
+      };
+    },
+  },
+  methods: {
+    togglePasswordVisibility() {
+      this.isPasswordVisible = !this.isPasswordVisible;
+    },
+    touchHandler(event) {
+      const isdateInputFocused = this.type === 'date' && event.type === 'focus';
+      const isDateInputBlured = this.type === 'date' && event.type === 'blur';
+      const isInputEmpty = !event.target.value;
+
+      if (isdateInputFocused) {
+        this.isdateInputText = true;
+      } else if (isDateInputBlured && isInputEmpty) {
+        this.isdateInputText = false;
+      }
+    },
+  },
```

Comment:
While all of this seems to work, keep in mind that all of this extra logic dealing with either password input types or date input types should -ideally- be outside of this component.
In the end this is the **Base**Input. You can create specific components (`PasswordInput`, `DateInput`) to handle specific cases which would use the BaseInput undernearth and properly configure it/expand upon it.

Reply:
Pretty much!
This will also make the component much more focused and be more "single-purpose" (remember the F.I.R.S.T. rules? 😄) which in turn - makes it more clean!
And if the password input would have let's imagine some more complex features - like I don't know in-site password-manager like thing - we would only extend the `PasswordInput` instead of building upon the ever-growing `BaseInput` 😄

Reply:
Sure! You can leave it for next PR 😄